src/ripple/app/paths/RippleState.cpp

```diff
@@ -25,58 +25,55 @@
 
 namespace ripple {
 
-RippleState::pointer
+RippleState::RippleState(
+    std::shared_ptr<SLE const> const& sle,
+    AccountID const& viewAccount)
+    : key_(sle->key())
+    , mLowLimit(sle->getFieldAmount(sfLowLimit))
+    , mHighLimit(sle->getFieldAmount(sfHighLimit))
+    , mBalance(sle->getFieldAmount(sfBalance))
+    , lowQualityIn_(sle->getFieldU32(sfLowQualityIn))
+    , lowQualityOut_(sle->getFieldU32(sfLowQualityOut))
+    , highQualityIn_(sle->getFieldU32(sfHighQualityIn))
+    , highQualityOut_(sle->getFieldU32(sfHighQualityOut))
+    , mFlags(sle->getFieldU32(sfFlags))
+    , mViewLowest(mLowLimit.getIssuer() == viewAccount)
+{
+    if (!mViewLowest)
+        mBalance.negate();
+}
+
+std::optional<RippleState>
 RippleState::makeItem(
     AccountID const& accountID,
-    std::shared_ptr<SLE const> sle)
+    std::shared_ptr<SLE const> const& sle)
 {
     // VFALCO Does this ever happen in practice?
```

Comment:
Please remove this comment, or at least update it:

It's possible that `sle == nullptr` but that would indicate an orphaned entry in an owner directory, which should, IMO, be caught elsewhere.

It's common to have `sle->getType() != ltRIPPLE_STATE` because we call this function as we iterate owner directories, for every entry in that directory, and entries of different types are expected.

Reply:
fixed in `0a3e192530 [fold] Rm old comments`